tests/src/python/test_provider_ogr_general.py

```diff
@@ -0,0 +1,980 @@
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Even Rouault'
+__date__ = '2016-04-11'
+__copyright__ = 'Copyright 2016, Even Rouault'
+# This will get replaced with a git SHA1 when you do a git archive
+__revision__ = '$Format:%H$'
+
+import os
+import shutil
+import sys
+import tempfile
+
+from qgis.core import Qgis, QgsVectorLayer, QgsVectorDataProvider, QgsWkbTypes, QgsFeatureRequest, QgsGeometry
+from qgis.PyQt.QtCore import QDate
+from qgis.testing import (
+    start_app,
+    unittest
+)
+from utilities import unitTestDataPath
+from osgeo import gdal, osr, ogr
+from inspect import getmembers
+from pprint import pprint
+
+start_app()
+TEST_DATA_DIR = unitTestDataPath()
+
+
+def GDAL_COMPUTE_VERSION(maj, min, rev):
+    return ((maj) * 1000000 + (min) * 10000 + (rev) * 100)
+
+# Note - doesn't implement ProviderTestCase as most OGR provider is tested by the shapefile provider test
+
+
+def count_opened_filedescriptors(filename_to_test):
+    count = -1
+    if sys.platform.startswith('linux'):
+        count = 0
+        open_files_dirname = '/proc/%d/fd' % os.getpid()
+        filenames = os.listdir(open_files_dirname)
+        for filename in filenames:
+            full_filename = open_files_dirname + '/' + filename
+            if os.path.exists(full_filename):
+                link = os.readlink(full_filename)
+                if os.path.basename(link) == os.path.basename(filename_to_test):
+                    count += 1
+    return count
+
+
+class TestPyQgsOGRProviderGeneral(unittest.TestCase):
+
+    ogr_spatialview_insert=0
+    ogr_spatialview_update=0
+    ogr_spatialview_delete=0
+    ogr_spatialview_selectatid=0
+    ogr_spatialview_changegeonetries=0
+    gdal_version_num=0
+    gdal_build_num=0
+    gdal_build_version=""
+    ogr_runtime_supported=1
+
+    def setUp(self):
+        """Run before each test."""
+        self.gdal_version_num = int(gdal.VersionInfo('VERSION_NUM'))
+        try:
+            Qgis.GDAL_BUILD_VERSION
+        except AttributeError:
+            self.gdal_build_num = self.gdal_version_num
+            self.gdal_build_version = gdal.VersionInfo('VERSION_NUM')
+        else:
+            self.ogr_runtime_supported = Qgis.GDAL_OGR_RUNTIME_SUPPORTED
+            self.gdal_build_version = Qgis.GDAL_BUILD_VERSION
+            self.gdal_build_num = (Qgis.GDAL_BUILD_VERSION_MAJOR*1000000)+(Qgis.GDAL_BUILD_VERSION_MINOR*10000)+(Qgis.GDAL_BUILD_VERSION_REV*100)
+
+
+    @classmethod
+    def setUpClass(cls):
+        """Run before all tests"""
+
+
+        # Create test layer
+        cls.basetestpath = tempfile.mkdtemp()
+        cls.dirs_to_cleanup = [cls.basetestpath]
+
+    @classmethod
+    def tearDownClass(cls):
+        """Run after all tests"""
+        for dirname in cls.dirs_to_cleanup:
+            shutil.rmtree(dirname, True)
+
```

Comment:
For GML, it was needed because the present code could not determine the Geometry-Type correctly
- always returned POINT

That is why one was added with 3 different Geometry-Types
* checked each type
   * checked if the first point is correct
      * with the accessing method to retrieve the first POINT being different